File: idioms/javascript/Closure.js
#! /usr/bin/env node
"use strict";

const CreateMethodWithContext = (value) => {
    const context = value;
    function Method() {
      return context;
    }
    return Method;
}

const origMethod = CreateMethodWithContext('origin value');
const newMethod = CreateMethodWithContext('new value');

console.log('context in orginal method: ' + origMethod());
console.log('context in new method: ' + newMethod());

File: idioms/python/Closure.py
#! /usr/bin/env python
import copy

def CreateMethodWithContext(value):
    context = copy.copy(value)
    def Method():
        return context
    return Method

orig_method = CreateMethodWithContext('original value')
new_method = CreateMethodWithContext('new value')

print('context in orginal method: ', orig_method())
print('context in new method: ', new_method())
